lib/vault/api/sys/namespace.rb

```diff
@@ -0,0 +1,81 @@
+module Vault
+  class Namespace < Response
+    # @!attribute [r] id
+    #   ID of the namespace
+    #   @return [String]
+    field :id
+
+    # @!attribute [r] path
+    #   Path of the namespace, includes parent paths if nested.
+    #   @return [String]
+    field :path
+  end
+
+  class Sys
+    # List all namespaces in a given scope. Ignores nested namespaces.
+    #
+    # @example
+    #   Vault.sys.namespaces #=> { :foo => #<struct Vault::Namespace id="xxxx1", path="foo/" }
+    #
+    #   @return [Hash<Symbol, Namespace>]
+    #
+    #   NOTE: Due to a bug in Vault, to be fixed soon, this method CAN return a pure JSON string if a scoping namespace is provided.
+    def namespaces(scoped=nil)
+      path = ["v1", scoped, "sys", "namespaces"].compact
+      json = client.list(path.join("/"))
+      json = json[:data] if json[:data]
+      if json[:key_info]
+        json = json[:key_info] if json[:key_info]
+        return Hash[*json.map do |k,v|
+          [k.to_s.chomp("/").to_sym, Namespace.decode(v)]
+        end.flatten]
```

Comment:
This is a dense piece of code; When will the `if` on line 28 be false? When is `flatten` called here, is it called on the result of the lambda? And the result of that is then sent to `Hash`?

Reply:
I pulled this directly from elsewhere in the codebase. I agree that it's super dense. I'll make it easier to read. :)

Reply:
I appreciate the effort of being consistent with other parts of the code! That said, the changes made in 89a18c358abf1abbf43ed657ee6b7eb94c418e6f are much easier to read and reason about 😄